packages/plugin-core/src/components/views/PreviewViewFactory.ts

```diff
@@ -28,6 +33,17 @@ export interface PreviewProxy {
    * @param note Note Props to update the preview contents with
    */
   updateForNote(note: NoteProps): void;
+
+  /**
```

Comment:
The Proxy was meant only as a temporary interface until we can refactor out other usages of 'updateForNote'.  Let's not further expose the panel by adding these methods.

Right now, we're still in the middle of refactoring the panel dependency issues, so it's hard to properly get an instance of the _panel properly. 

Rather than doing the `ConfigUtils.getPreview(config).automaticallyShowPreview` inside this method, can we do something like 

_in the code where we want to open the preview:_

```typescript
if (ConfigUtils.getPreview(config).automaticallyShowPreview) {
  const ws = getExtension();
  await new ShowPreviewCommand(PreviewPanelFactory.create(ws)).run();
}
```

I think this should have the same effect? 

Let me know if it's helpful to discuss further offline.

Reply:
1. Unfortunately calling ShowPreviewCommand is not enough as I should only call if the panel does not exist. Otherwise it leads to a strange visual side effect where the preview panel is always in effect (maybe easier to show than tell). I updated the existing method you exposed to do both show preview and update, moving the update to a private method of the factory. This way only one method gets exposed instead of two.

2. I need the _panel property for testing purposes. Is there another way to check if the panel exists?

Reply:
Ah - I see.  

Re point 2 - Maybe instead of exposing the panel directly, we can expose a property on PreviewProxy like `PanelVisible` or something like that.  My main worry is losing encapsulation on the preview panel again. Let's defer that change to next time.